packages/host/app/services/matrix-service.ts

```diff
@@ -150,9 +150,12 @@ export default class MatrixService extends Service {
     () => this.cardAPI,
   );
 
+  // TODO This seems very bad. we should not be sharing Resources with anyone that
+  // wants one--resources are tied to the lifetime of their owner, who knows
+  // which owner made these and who is consuming these. we need to refactor this out..
   roomResourcesCache: TrackedMap<string, RoomResource> = new TrackedMap();
```

Comment:
we need to stop using resources like this

Reply:
what is the pattern to handle something like this? 

I guess this shudnt even be in the matrix service in the first place

Reply:
confirm.

Reply:
if we need to have a cache it should be at the resource level, like how the store caches cards for card resources. but it does mean managing resource lifetime for the cache